utils/compile-time-utils/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+#[macro_export]
+macro_rules! define_version_string {
+    () => {
+        #[macro_use]
+        extern crate lazy_static;
+
+        lazy_static! {
```

Comment:
I think it would be more readable and usable to define it like [`clap::crate_version`](https://github.com/clap-rs/clap/blob/81457178fa7e055775867ca659b37798b5ae9584/src/macros.rs#L34-L55) which enables using this inplace.

I think lazy static is not adding much value here

Reply:
With `lazy_static` I can make the definition of the version string conditional: it's `"${CARGO_PKG_VERSION}-${GITHUB_SHA}"` if `GITHUB_SHA` is defined and `"${CARGO_PKG_VERSION}"` otherwise. I don't know how to do this using only macros.